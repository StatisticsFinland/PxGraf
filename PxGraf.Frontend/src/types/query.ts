/* istanbul ignore file */

/* Query - dimension queries*/

/**
 * Type for a query that contains @see {@link IDimensionQuery} objects for each dimension.
 * @property {{ [key: string]: IDimensionQuery }} - Queries for each dimension.
 */
export type Query = { [key: string]: IDimensionQuery };

/**
 * Enum that defines different dimension value filter types.
 */
export enum FilterType {
    Item = 'item',
    All = 'all',
    Top = 'top',
    From = 'from'
}

/**
 * Interface for dimension query properties.
 * @property {IValueFilter} valueFilter - Reference to value filter.
 * @property {boolean} selectable - Selected selectable dimension.
 * @property {any} virtualValueDefinitions - Virtual value definitions.
  */
export interface IDimensionQuery {
    valueFilter: IValueFilter
    selectable: boolean
    virtualValueDefinitions
}

/**
 * Interface for value filter properties.
 * @property {FilterType} type - Selected filter type.
 * @property {string[] | string | number} query - The query for the filter.
 */
export interface IValueFilter {
    type: FilterType;
    query?: string[] | string | number
}

/**
 * Type for string supporting multiple languages.
 */
type MultiLanguageString = { [key: string]: string };

/* CubeQuery - meta data editions*/

/**
 * Interface for cube query properties.
 * @property {MultiLanguageString} chartHeaderEdit - The chart header edition.
 * @property {{ [key: string]: IDimensionEditions }} variableQueries - The IDimensionEditions edition queries.
 */
export interface ICubeQuery {
    chartHeaderEdit?: MultiLanguageString;
    variableQueries: { [key: string]: IDimensionEditions };
}

/**
 * Interface for IDimensionEditions editions.
 * @property {{ [key: string]: IDimensionValueEditions }} valueEdits - The value editions.
 */
export interface IDimensionEditions {
    valueEdits: {
        [valueCode: string]: IDimensionValueEditions;
    };
}

/**
 * Interface for dimension value editions.
 * @property {MultiLanguageString} nameEdit - The name edition.
 * @property {IContentComponentEdition} contentComponent - The content component edition containing source and unit.
  */
export interface IDimensionValueEditions {
    nameEdit? : MultiLanguageString;
    contentComponent? : IContentComponentEdition;
}

/**
 * Interface for content component edition.
 * @property {MultiLanguageString} sourceEdit - Edition of the source.
 * @property {MultiLanguageString} unitEdit - Edition of the unit.
 */
interface IContentComponentEdition {
    sourceEdit? : MultiLanguageString;
    unitEdit? : MultiLanguageString;
}

/**
 * Interface that stores information about the query.
 * @property {number} size - The size of the query.
 * @property {number} sizeWarningLimit - The size warning limit of the query.
 * @property {number} maximumSupportedSize - The maximum supported size of the query.
 * @property {string[]} validVisualizations - Names of valid visualization types for of the query.
 * @property {{[key: string]: MultiLanguageString}} visualizationRejectionReasons - Reasons for why certain visualization types are rejected for the query.
 * @property {number} maximumHeaderLength - The maximum header length of the query.
 */
export interface IQueryInfo {
    size: number;
    sizeWarningLimit: number;
    maximumSupportedSize: number;
    validVisualizations: string[];
    visualizationRejectionReasons: {[key: string]: MultiLanguageString};
    maximumHeaderLength: number;
}