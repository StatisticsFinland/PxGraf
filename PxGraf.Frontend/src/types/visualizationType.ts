/* istanbul ignore file */

/**
 * Enum that defines different visualization types.
 */
export enum VisualizationType {
    VerticalBarChart = 'VerticalBarChart',
    GroupVerticalBarChart = 'GroupVerticalBarChart',
    StackedVerticalBarChart = 'StackedVerticalBarChart',
    PercentVerticalBarChart = 'PercentVerticalBarChart',
    HorizontalBarChart = 'HorizontalBarChart',
    GroupHorizontalBarChart = 'GroupHorizontalBarChart',
    StackedHorizontalBarChart = 'StackedHorizontalBarChart',
    PercentHorizontalBarChart = 'PercentHorizontalBarChart',
    PyramidChart = 'PyramidChart',
    PieChart = 'PieChart',
    LineChart = 'LineChart',
    ScatterPlot = 'ScatterPlot',
    Table = 'Table'
}