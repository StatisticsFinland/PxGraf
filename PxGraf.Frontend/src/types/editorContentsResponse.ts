import { MultiLanguageString } from "./multiLanguageString";
import { VisualizationType } from "./visualizationType";

/**
 * Interface for dimension sorting options.
 * @property {string} code - The code of the sorting option.
 * @property {MultiLanguageString} description - Longer, user facing description of the sorting option.
 */
export interface ISortingOption {
    code: string;
    description: MultiLanguageString;
}

/**
 * Interface for sorting option collection. Contains default and pivoted sorting options if available.
 * @property {ISortingOption[]} default - Default sorting options.
 * @property {ISortingOption[]} pivoted - Pivoted sorting options.
 */
interface ISortingOptionCollection {
    default: ISortingOption[];
    pivoted: ISortingOption[];
}

/**
 * Interface for visualization options for a specific visualization type.
 * @property {VisualizationType} type - The type of visualization.
 * @property {boolean} [allowShowingDataPoints] - Flag to indicate if showing data points is allowed.
 * @property {boolean} [allowCuttingYAxis] - Flag to indicate if cutting the Y-axis is allowed.
 * @property {boolean} [allowMatchXLabelsToEnd] - Flag to indicate if matching X labels to the end is allowed.
 * @property {boolean} [allowSetMarkerScale] - Flag to indicate if setting marker scale is allowed.
 * @property {boolean} allowManualPivot - Flag to indicate if manual pivoting is allowed.
 * @property {boolean} allowMultiselect - Flag to indicate if multivalue selectable is allowed.
 * @property {ISortingOptionCollection} sortingOptions - Sorting options for the visualization containing default and pivoted options.
 */
export interface IVisualizationOptions {
    type: VisualizationType;
    allowShowingDataPoints?: boolean;
    allowCuttingYAxis?: boolean;
    allowMatchXLabelsToEnd?: boolean;
    allowSetMarkerScale?: boolean;
    allowManualPivot: boolean;
    allowMultiselect: boolean;
    sortingOptions: ISortingOptionCollection;
}

/**
 * Interface for the response from the editor contents API used to populate the settings for adjusting the visualization in the editor.
 * @property {number} size - The size of the response based on the amount of values selected from all dimensions.
 * @property {number} sizeWarningLimit - The size warning limit for the response.
 * @property {number} maximumSupportedSize - The maximum supported size for the response.
 * @property {MultiLanguageString} headerText - Default header text for the visualization as a @ {@link MultiLanguageString} object.
 * @property {number} maximumHeaderLength - The maximum length of the header text.
 * @property {IVisualizationOptions[]} visualizationOptions - Visualization options for the valid visualization types as an array of @ {@link IVisualizationOptions} objects.
 * @property {object} visualizationRejectionReasons - Object containing rejection reasons for each visualization type as a @ {@link MultiLanguageString} object.
 */
export interface IEditorContentsResponse {
    size: number;
    sizeWarningLimit: number;
    maximumSupportedSize: number;
    headerText: MultiLanguageString;
    maximumHeaderLength: number;
    visualizationOptions: IVisualizationOptions[];
    visualizationRejectionReasons: {
        [visualizationType: string]: MultiLanguageString;
    }
}