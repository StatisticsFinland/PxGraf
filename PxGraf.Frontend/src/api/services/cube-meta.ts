/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { ICubeMeta } from "types/cubeMeta";
import { defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for cube metadata result. Cube is a multi-dimensional result of Px file data filtered by a query.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {ICubeMeta} data - The cube meta data as an @see {@link ICubeMeta} object.
 */
export interface ICubeMetaResult {
    isLoading: boolean;
    isError: boolean;
    data: ICubeMeta;
}

const fetchCubeMeta = async(idStack: string[]): Promise<ICubeMeta> => {
    const client = new ApiClient();
    const url = `creation/cube-meta/${idStack.join('/')}`;

    return await client.getAsync(url);
}

export const useCubeMetaQuery = (idStack: string[]): ICubeMetaResult => {
    return useQuery(
        ['cube-meta', idStack],
        () => fetchCubeMeta(idStack),
        defaultQueryOptions
    );
}