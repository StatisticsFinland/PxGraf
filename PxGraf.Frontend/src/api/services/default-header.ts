/* istanbul ignore file */

import ApiClient from "api/client";
import { merge } from "lodash";
import { useQuery } from "react-query";
import { MultiLanguageString } from "types/multiLanguageString";
import { Query } from "types/query";
import { buildTableReference, defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for the chart's header result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred while retrieving the header.
 * @property {MultiLanguageString} data - The header data represented as a multi-language string.
 */
export interface IHeaderResult {
    isLoading: boolean;
    isError: boolean;
    data: MultiLanguageString;
}

const fetchDefaultHeader = async(idStack: string[], query: Query): Promise<MultiLanguageString> => {
    const client = new ApiClient();
    const requestBody = JSON.stringify(
        merge(
            { tableReference: buildTableReference(idStack) },
            { variableQueries: query }
        )
    );
  
    const url = 'creation/default-header';
    return await client.postAsync(url, requestBody);
}

export const useDefaultHeaderQuery = (idStack: string[], query: Query): IHeaderResult => {
    return useQuery(
        ['default-header', ...idStack, query],
        () => fetchDefaultHeader(idStack, query),
        {
            ...defaultQueryOptions,
            enabled: query != null,
        }
    );
}