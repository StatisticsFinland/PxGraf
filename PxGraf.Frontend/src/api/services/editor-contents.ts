/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { IEditorContentsResponse } from "types/editorContentsResponse";
import { buildCubeQuery, defaultQueryOptions, useDebounceState } from "utils/ApiHelpers";
import { ICubeQuery, Query } from "types/query";

/**
 * Interface for editor contents result. Editor contents are used by the editor to populate the settings for adjusting the visualization.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {IEditorContentsResponse} data - The editor contents as an @see {@link IEditorContentsResponse} object.
 */ 
export interface IEditorContentsResult {
    isLoading: boolean;
    isError: boolean;
    data: IEditorContentsResponse;
}

const fetchEditorContents = async (idStack: string[], query: Query, metaEdits: ICubeQuery): Promise<IEditorContentsResponse> => {
    const client = new ApiClient();
    const url = 'creation/editor-contents';
    const requestBody = JSON.stringify(buildCubeQuery(query, metaEdits, idStack));
    return await client.postAsync(url, requestBody);
}

export const useEditorContentsQuery = (idStack: string[], query: Query, cubeQuery: ICubeQuery): IEditorContentsResult => {
    [idStack, query, cubeQuery] = useDebounceState(1000, idStack, query, cubeQuery);
    const queryKey = ['editor-contents', ...idStack, query, cubeQuery];
    return useQuery(
        queryKey,
        () => fetchEditorContents(idStack, query, cubeQuery),
        {
            ...defaultQueryOptions,
            enabled: query != null,
            keepPreviousData: true,
        }
    );
}