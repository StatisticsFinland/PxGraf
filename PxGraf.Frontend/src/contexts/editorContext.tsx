/* istanbul ignore file */

import { debounce } from 'lodash';
import * as React from 'react';
import { ICubeQuery, Query } from 'types/query';
import { IVisualizationSettings } from 'types/visualizationSettings';
import { VisualizationType } from 'types/visualizationType';

/**
 * Interface for editor context properties
 * @property {ICubeQuery} cubeQuery - Query object that contains the header and dimension queries
 * @property {React.Dispatch<ICubeQuery>} setCubeQuery - Function to set the cube query
 * @property {Query} query - Query object that contains the dimension queries
 * @property {React.Dispatch<Query>} setQuery - Function to set the query
 * @property {boolean} saveDialogOpen - Flag to indicate if the save dialog is open
 * @property {React.Dispatch<boolean>} setSaveDialogOpen - Function to set the save dialog open flag
 * @property {VisualizationType} selectedVisualizationUserInput - The selected visualization type
 * @property {React.Dispatch<VisualizationType>} setSelectedVisualizationUserInput - Function to set the selected visualization type
 * @property {IVisualizationSettings} visualizationSettingsUserInput - The visualization settings
 * @property {React.Dispatch<IVisualizationSettings>} setVisualizationSettingsUserInput - Function to set the visualization settings
 * @property {{ [key: string]: string[] }} defaultSelectables - The default selectables for each dimension
 * @property {React.Dispatch<{ [key: string]: string[] }>} setDefaultSelectables - Function to set the default selectables
 */
interface IEditorContext {
    cubeQuery: ICubeQuery;
    setCubeQuery: React.Dispatch<ICubeQuery>;
    query: Query;
    setQuery: React.Dispatch<Query>;
    saveDialogOpen: boolean;
    setSaveDialogOpen: React.Dispatch<boolean>;
    selectedVisualizationUserInput: VisualizationType;
    setSelectedVisualizationUserInput: React.Dispatch<VisualizationType>;
    visualizationSettingsUserInput: IVisualizationSettings;
    setVisualizationSettingsUserInput: React.Dispatch<IVisualizationSettings>;
    defaultSelectables: { [key: string]: string[] };
    setDefaultSelectables: React.Dispatch<{ [key: string]: string[] }>
}

/**
 * Context required for the editor
 */
export const EditorContext = React.createContext<IEditorContext>({
    cubeQuery: null,
    setCubeQuery: () => { /* no base implementation */ },
    query: null,
    setQuery: () => { /* no base implementation */ },
    saveDialogOpen: false,
    setSaveDialogOpen: () => { /* no base implementation */ },
    selectedVisualizationUserInput: null,
    setSelectedVisualizationUserInput: () => { /* no base implementation */ },
    visualizationSettingsUserInput: null,
    setVisualizationSettingsUserInput: () => { /* no base implementation */ },
    defaultSelectables: null,
    setDefaultSelectables: () => { /* no base implementation */ }
});

/**
 * Provider for the editor context
 * @param children - The child components
 */
export const EditorProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
    const [cubeQuery, setCubeQueryState] = React.useState({ variableQueries: {} });
    const [query, setQuery] = React.useState(null);
    const [saveDialogOpen, setSaveDialogOpen] = React.useState(false);
    const [selectedVisualizationUserInput, setSelectedVisualizationUserInput] = React.useState(null);
    const [visualizationSettingsUserInput, setVisualizationSettingsUserInput] = React.useState(null);
    const [defaultSelectables, setDefaultSelectables] = React.useState(null);

    const debouncedSetCubeQuery = React.useMemo(() => {
        return debounce((newQuery: ICubeQuery) => {
            setCubeQueryState(newQuery);
        }, 1000);
    }, []);

    const setCubeQuery = React.useCallback((newQuery: ICubeQuery) => {
        debouncedSetCubeQuery(newQuery);
    }, [debouncedSetCubeQuery]);

    React.useEffect(() => {
        return () => {
            debouncedSetCubeQuery.cancel();
        };
    }, [debouncedSetCubeQuery]);

    const contextValue = React.useMemo(() => {
        return {
            cubeQuery, setCubeQuery,
            query, setQuery,
            saveDialogOpen, setSaveDialogOpen,
            selectedVisualizationUserInput, setSelectedVisualizationUserInput,
            visualizationSettingsUserInput, setVisualizationSettingsUserInput,
            defaultSelectables, setDefaultSelectables
        }
    }, [
        cubeQuery, query, saveDialogOpen, selectedVisualizationUserInput,
        visualizationSettingsUserInput, defaultSelectables
    ]);

    return (
        <EditorContext.Provider value={contextValue}>
            {children}
        </EditorContext.Provider>
    );
}