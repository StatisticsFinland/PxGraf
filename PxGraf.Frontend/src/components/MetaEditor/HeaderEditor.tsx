import React from 'react';
import { useTranslation } from 'react-i18next';
import { Alert, Skeleton } from '@mui/material';
import { EditorField } from './Editorfield';
import { IHeaderResult } from 'api/services/default-header';
import InfoBubble from 'components/InfoBubble/InfoBubble';
import styled from 'styled-components';
import { EditorContext } from '../../contexts/editorContext';
import { MultiLanguageString } from '../../types/multiLanguageString';

interface IHeaderEditorProps {
    defaultHeaderResponse: IHeaderResult;
    language: string;
    style: { [key: string]: string | number }
    maxLength?: number;
}

const Wrapper = styled.div`
  display: flex;
`;

const GridFixer = styled.div`
  grid-column: span 12;
`;

export const HeaderEditor: React.FC<IHeaderEditorProps> = ({ defaultHeaderResponse, language, maxLength, style = {} }) => {
    const { t } = useTranslation();

    const { cubeQuery, setCubeQuery } = React.useContext(EditorContext);
    const editValue = cubeQuery?.chartHeaderEdit;
    const editHeader = (title: MultiLanguageString) => setCubeQuery({ ...cubeQuery, chartHeaderEdit: title })

    if (defaultHeaderResponse.isError) {
        return (
            <GridFixer>
                <Alert style={style} severity="error">{t("error.contentLoad")}</Alert>
            </GridFixer>
        );
    }
    else if (defaultHeaderResponse.isLoading) {
        return (
            <GridFixer>
                <Skeleton style={style} variant="rectangular" height={55} />
            </GridFixer>
        );
    }
    else {
        return (
            <GridFixer>
                <Wrapper>
                    <InfoBubble info={t('infoText.titleEdition')} ariaLabel={t("editMetadata.header")} />
                    <EditorField
                        label={t("editMetadata.header")}
                        style={style}
                        defaultValue={defaultHeaderResponse.data[language]}
                        editValue={editValue ? editValue[language] : null}
                        onChange={newValue => {
                            const newEdit = { ...editValue, [language]: newValue };
                            editHeader(newEdit);
                        }}
                        maxLength={maxLength}
                    />
                </Wrapper>
            </GridFixer>
        );
    }
}

export default HeaderEditor;