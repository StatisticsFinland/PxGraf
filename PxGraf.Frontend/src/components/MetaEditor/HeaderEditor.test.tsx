import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import HeaderEditor from './HeaderEditor';
import { EditorContext } from '../../contexts/editorContext';
import { VisualizationType } from '../../types/visualizationType';
import { ICubeQuery } from '../../types/query';
import { IEditorContentsResult } from '../../api/services/editor-contents';
import { IEditorContentsResponse } from '../../types/editorContentsResponse';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

const data: IEditorContentsResponse = {
    headerText:
    {
        'fi': 'foo'
    }
} as unknown as IEditorContentsResponse;

const mockDefaultResponse: IEditorContentsResult = {
    isLoading: false,
    isError: false,
    data: data
};

const mockDefaultResponseLoading: IEditorContentsResult = {
    isLoading: true,
    isError: false,
    data: data
};

const mockDefaultResponseError: IEditorContentsResult = {
    isLoading: false,
    isError: true,
    data: data
};

const mockLang = 'fi';
const mockFunction = jest.fn();
const mockCubeQuery: ICubeQuery = {
    chartHeaderEdit: { 'fi': 'bar' },
    variableQueries: {}
}

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <EditorContext.Provider value={{
                defaultSelectables: {},
                setDefaultSelectables: jest.fn(),
                cubeQuery: mockCubeQuery,
                setCubeQuery: mockFunction,
                query: {},
                setQuery: jest.fn(),
                saveDialogOpen: false,
                setSaveDialogOpen: jest.fn(),
                selectedVisualizationUserInput: VisualizationType.VerticalBarChart,
                setSelectedVisualizationUserInput: jest.fn(),
                visualizationSettingsUserInput: {},
                setVisualizationSettingsUserInput: jest.fn()
            }}>
                <HeaderEditor editorContentResponse={mockDefaultResponse} language={mockLang} style={{}} />
            </EditorContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly when loading', () => {
        const { asFragment } = render(
                <EditorContext.Provider value={{
                    defaultSelectables: {},
                    setDefaultSelectables: jest.fn(),
                    cubeQuery: mockCubeQuery,
                    setCubeQuery: mockFunction,
                    query: {},
                    setQuery: jest.fn(),
                    saveDialogOpen: false,
                    setSaveDialogOpen: jest.fn(),
                    selectedVisualizationUserInput: VisualizationType.VerticalBarChart,
                    setSelectedVisualizationUserInput: jest.fn(),
                    visualizationSettingsUserInput: {},
                    setVisualizationSettingsUserInput: jest.fn()
                }}>
                    <HeaderEditor editorContentResponse={mockDefaultResponseLoading} language={mockLang} style={{}} />
                </EditorContext.Provider>

        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly when receiving error', () => {
        const { asFragment } = render(
                    <EditorContext.Provider value={{
                        defaultSelectables: {},
                        setDefaultSelectables: jest.fn(),
                        cubeQuery: mockCubeQuery,
                        setCubeQuery: mockFunction,
                        query: {},
                        setQuery: jest.fn(),
                        saveDialogOpen: false,
                        setSaveDialogOpen: jest.fn(),
                        selectedVisualizationUserInput: VisualizationType.VerticalBarChart,
                        setSelectedVisualizationUserInput: jest.fn(),
                        visualizationSettingsUserInput: {},
                        setVisualizationSettingsUserInput: jest.fn()
                    }}>
                        <HeaderEditor editorContentResponse={mockDefaultResponseError} language={mockLang} style={{}} />
                    </EditorContext.Provider>

        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Assertion tests', () => {
    it('Change event should fire when value has changed', () => {
        render(
            <EditorContext.Provider value={{
                defaultSelectables: {},
                setDefaultSelectables: jest.fn(),
                cubeQuery: mockCubeQuery,
                setCubeQuery: mockFunction,
                query: {},
                setQuery: jest.fn(),
                saveDialogOpen: false,
                setSaveDialogOpen: jest.fn(),
                selectedVisualizationUserInput: VisualizationType.VerticalBarChart,
                setSelectedVisualizationUserInput: jest.fn(),
                visualizationSettingsUserInput: {},
                setVisualizationSettingsUserInput: jest.fn()
            }}>
                <HeaderEditor editorContentResponse={mockDefaultResponse} language={mockLang} style={{}} />
            </EditorContext.Provider>
        );
        fireEvent.change(screen.getByDisplayValue(mockCubeQuery.chartHeaderEdit['fi']), { target: { value: 'editValue2' } });
        expect(mockFunction).toHaveBeenCalledTimes(1);
    });
});
