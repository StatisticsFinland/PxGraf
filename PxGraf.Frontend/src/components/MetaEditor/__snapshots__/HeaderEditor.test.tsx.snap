// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-jqVXSH kGpIkX"
  >
    <div
      class="sc-dDvxFM diyMsl"
    >
      <button
        aria-label="tooltip.open: editMetadata.header"
        class="sc-gyycJP jHfhmG"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
          />
        </svg>
      </button>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
        >
          <b>
            editMetadata.header*
          </b>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd sc-KXCwU lmZjwF css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            type="text"
            value="bar"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <span
              aria-label="editMetadata.discardChanges"
              class=""
              data-mui-internal-clone-element="true"
            >
              <button
                aria-label="editMetadata.discardChanges"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="UndoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </span>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                editMetadata.header
              </span>
            </legend>
          </fieldset>
        </div>
        <div
          aria-live="polite"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Rendering test renders correctly when loading 1`] = `
<DocumentFragment>
  <div
    class="sc-jqVXSH kGpIkX"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
      style="height: 55px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Rendering test renders correctly when receiving error 1`] = `
<DocumentFragment>
  <div
    class="sc-jqVXSH kGpIkX"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-x7o5da-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        error.contentLoad
      </div>
    </div>
  </div>
</DocumentFragment>
`;
